### PR TITLE
Embed the full Debian package version number and fix the 5.7 build

### PR DESCRIPTION
**Pull request by tianon**
This fixes the issue in https://github.com/docker-library/mysql/commit/617c6470dfd96839a697d262eafef5242b9ea492#commitcomment-13911981 and https://travis-ci.org/docker-library/mysql/jobs/86708981#L245-L262, and gives us more resilient+consistent builds overall.

### CODE REVIEW

File: 5.7/Dockerfile [hunk 1]
```diff
@@ -16,9 +16,9 @@ RUN apt-get update && apt-get install -y perl --no-install-recommends && rm -rf
 RUN apt-key adv --keyserver ha.pool.sks-keyservers.net --recv-keys A4A9406876FCBD3C456770C88C718D3B5072E1F5
 
 ENV MYSQL_MAJOR 5.7
-ENV MYSQL_VERSION 5.7.9
+ENV MYSQL_VERSION 5.7.9-1debian8
 
-RUN echo "deb http://repo.mysql.com/apt/debian/ jessie mysql-${MYSQL_MAJOR}-dmr" > /etc/apt/sources.list.d/mysql.list
+RUN echo "deb http://repo.mysql.com/apt/debian/ jessie mysql-${MYSQL_MAJOR}" > /etc/apt/sources.list.d/mysql.list
```
Comment (tianon) [inline 5.7/Dockerfile]: This was the key for making it work now that 5.7's GA. :+1:

Comment (tianon) [inline 5.7/Dockerfile]: (https://dev.mysql.com/doc/mysql-development-cycle/en/development-milestone-releases.html)

**Comment by tianon** [comment]
Updated generate output:
```
5.5.46: git://github.com/docker-library/mysql@e892e8c52485535cce5dbadaba2c579fcc805f21 5.5
5.5: git://github.com/docker-library/mysql@e892e8c52485535cce5dbadaba2c579fcc805f21 5.5

5.6.27: git://github.com/docker-library/mysql@7e712faa54d78f30cb76ab3231fe8bc8c56e0143 5.6
5.6: git://github.com/docker-library/mysql@7e712faa54d78f30cb76ab3231fe8bc8c56e0143 5.6

5.7.9: git://github.com/docker-library/mysql@7e712faa54d78f30cb76ab3231fe8bc8c56e0143 5.7
5.7: git://github.com/docker-library/mysql@7e712faa54d78f30cb76ab3231fe8bc8c56e0143 5.7
5: git://github.com/docker-library/mysql@7e712faa54d78f30cb76ab3231fe8bc8c56e0143 5.7
latest: git://github.com/docker-library/mysql@7e712faa54d78f30cb76ab3231fe8bc8c56e0143 5.7
```

**Comment by yosifkit** [comment]
LGTM

**Pull request merged**